### PR TITLE
add schema file

### PR DESCRIPTION
**Pull request by vaniisgh**
As explained in #25, an API Service should be created to make the TEStribute deployable.

A rough implementation of an OpenAPI 3 schema that could be used for the service mentioned has been included in this pull request.

### CODE REVIEW

File: specs/schema.TEStribute.openapi.yaml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    post:
+      operationId: rank_services
+      description: Creates a new pet in the store.  Duplicates are allowed
+      requestBody:
+        description: Pet to add to the store
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/rank_services_requirements'
```
Comment (uniqueg) [inline specs/schema.TEStribute.openapi.yaml]: Could you please put the names in CamelCase or camelCase as it seems to be a convention for models, or not? Does OpenAPI 3.0 guide says anything about it? If you replace, please do consistently for all models

File: specs/schema.TEStribute.openapi.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      responses:
+        '200':
+          description: ranking successful
```
Comment (uniqueg) [inline specs/schema.TEStribute.openapi.yaml]: Start with capital letter

File: specs/schema.TEStribute.openapi.yaml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  license:
+    name: Apache 2.0
+    url: 'https://github.com/elixir-europe/TEStribute/blob/dev/LICENSE'
+paths:
+  /rank_services:
+    post:
+      operationId: rank_services
+      description: Creates a new pet in the store.  Duplicates are allowed
+      requestBody:
+        description: Pet to add to the store
```
Comment (uniqueg) [inline specs/schema.TEStribute.openapi.yaml]: Could be something like "Input parameters"

File: specs/schema.TEStribute.openapi.yaml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    name: 'Elixir Europe'
+    email: 'alexander.kanitz@alumni.ethz.ch'
+  license:
+    name: Apache 2.0
+    url: 'https://github.com/elixir-europe/TEStribute/blob/dev/LICENSE'
+paths:
+  /rank_services:
+    post:
+      operationId: rank_services
+      description: Creates a new pet in the store.  Duplicates are allowed
```
Comment (uniqueg) [inline specs/schema.TEStribute.openapi.yaml]: You can take this from TEStribute itself, there are good descriptions

File: specs/schema.TEStribute.openapi.yaml [hunk 5]
```diff
@@ -0,0 +1,96 @@
+openapi: "3.0.0"
+info:
+  version: 0.0.1
+  title: TEStribute OpenAPI specification
+  contact:
+    name: 'Elixir Europe'
+    email: 'alexander.kanitz@alumni.ethz.ch'
+  license:
+    name: Apache 2.0
+    url: 'https://github.com/elixir-europe/TEStribute/blob/dev/LICENSE'
```
Comment (uniqueg) [inline specs/schema.TEStribute.openapi.yaml]: I think better link to the official Apache 2.0 license at https://www.apache.org/licenses/LICENSE-2.0

File: specs/schema.TEStribute.openapi.yaml [hunk 6]
```diff
@@ -0,0 +1,96 @@
+openapi: "3.0.0"
+info:
+  version: 0.0.1
+  title: TEStribute OpenAPI specification
+  contact:
+    name: 'Elixir Europe'
```
Comment (uniqueg) [inline specs/schema.TEStribute.openapi.yaml]: Put here: ELIXIR Cloud & AAI group

File: specs/schema.TEStribute.openapi.yaml [hunk 7]
```diff
@@ -0,0 +1,96 @@
+openapi: "3.0.0"
+info:
+  version: 0.0.1
```
Comment (uniqueg) [inline specs/schema.TEStribute.openapi.yaml]: I think we tie this to the TEStribute version. If we don't have one, let's say 0.1.0 :)

File: specs/schema.TEStribute.openapi.yaml [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+    rank_services_requirements:
+      type: object
+      properties:
+        drs_ids:
+          $ref: '#/components/schemas/drs_ids'
+        drs_uris:
+          $ref: '#/components/schemas/drs_uris'
+        mode:
+          anyOf:
+            - type: integer
```
Comment (uniqueg) [inline specs/schema.TEStribute.openapi.yaml]: but should be enum of -1, 0, 1

File: specs/schema.TEStribute.openapi.yaml [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+            - type: string
```
Comment (uniqueg) [inline specs/schema.TEStribute.openapi.yaml]: should be enum of 'cost', 'time' and 'random' (I think, check the code)

File: specs/schema.TEStribute.openapi.yaml [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/rank_services_response'
+
+components:
+  schemas:
+    rank_services_requirements:
+      type: object
+      properties:
```
Comment (uniqueg) [inline specs/schema.TEStribute.openapi.yaml]: Please add descriptions everywhere, you can take them from the code

File: specs/schema.TEStribute.openapi.yaml [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+        zones:
+          type: array
+          items:
+            type: string
+    tes_uris:
+      type: array
+      items:
+        type: string
+    rank_services_response:
+      type: object
```
Comment (uniqueg) [inline specs/schema.TEStribute.openapi.yaml]: Isn't this an array of objects?

File: specs/schema.TEStribute.openapi.yaml [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+      properties:
```
Comment (uniqueg) [inline specs/schema.TEStribute.openapi.yaml]: Aren't there some properties missing? Like estimated costs, times

File: specs/schema.TEStribute.openapi.yaml [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+          $ref: '#/components/schemas/tes_drs_combinations'
+    tes_drs_combinations:
+      type: object
+      additionalProperties:
+          $ref: '#/components/schemas/objid_drs_combinations'
+    objid_drs_combinations:
+      type: object
+      properties:
+        object_id:
+          additionalProperties:
```
Comment (uniqueg) [inline specs/schema.TEStribute.openapi.yaml]: What's this?

Comment (vaniisgh) [reply specs/schema.TEStribute.openapi.yaml]: I tried to nest them in like dictionaries, not sure how that works honestly . I looked at it here https://swagger.io/docs/specification/data-models/dictionaries/.

Comment (vaniisgh) [reply specs/schema.TEStribute.openapi.yaml]: I think I'll change this completely though, I think this is too complicated ...

File: specs/schema.TEStribute.openapi.yaml (hunk 8 above)
Comment (vaniisgh) [reply specs/schema.TEStribute.openapi.yaml]: I think you've modified this to allow quite a few options in the `validate_inputs` module. I might have understood incorrectly I will check and modify accordingly.

**Comment by uniqueg** [comment]
Okay, cool - that's done. But will require a few minor changes in the TEStribute code (I wrote the specs how the input and return objects _should be_, not how they are ;-)

**Pull request merged**